scu_builders.py

```diff
@@ -0,0 +1,335 @@
+"""Functions used to generate scu build source files during build time
+"""
+import glob, os
+import math
+from pathlib import Path
+from os.path import normpath, basename
+
+base_folder = os.path.abspath("./") + "/"
+_verbose = False
+_is_release_build = False
+
+
+def clear_out_existing_files(output_folder, extension):
+    output_folder = os.path.abspath(output_folder)
+    # print("clear_out_existing_files from folder: " + output_folder)
+
+    if not os.path.isdir(output_folder):
+        # folder does not exist or has not been created yet,
+        # no files to clearout. (this is not an error)
+        return
+
+    os.chdir(output_folder)
+
+    for file in glob.glob("*." + extension):
+        # print("removed pre-existing file: " + file)
+        os.remove(file)
+
+
+def folder_not_found(folder):
+    abs_folder = base_folder + folder + "/"
+    return os.path.isdir(abs_folder) == False
+
+
+def find_files_in_folder(folder, sub_folder, include_list, extension, sought_exceptions, found_exceptions):
+    abs_folder = base_folder + folder + "/" + sub_folder
```

Comment:
much simple with `pathlib.Path`:
```suggestion
    abs_folder = base_folder / folder / sub_folder
```